src/python/vdo-setup-dmtest/vdo-setup-dmtest.py

```diff
@@ -0,0 +1,512 @@
+
+  def runRepoInstallSteps(self):
+    """
+    Run the correpsonding install method for each repository that was cloned
+    """
+    self.log.info("\nRunning the install methods for each repo")
+    for key in self.repoKeys:
+      repoName = self.config[key].rpartition("/")[-1].rsplit(".", 1)[0]
+      repoPath = self.config["_".join([repoName.upper(), 'PATH'])]
+
+      # Perform necessary installation steps for the repos
+      self.log.info("Installing {0}".format(repoName))
+      try:
+        os.chdir(repoPath)
+
+        # The repo keys are sorted, so install should go in this order
+        if (key.upper().find('BUFIO') != -1):
+          self.installBufioTest(git.Repo(repoPath))
+        elif (key.upper().find('BLK_ARCHIVE') != -1):
+          self.installBlkArchive(repoName)
+        elif (key.upper().find('DMTEST') != -1):
+          self.installDmtestPython(repoName)
+        else:
+          msg = 'does not match any of the expected substrings (BUFIO | BLK_ARCHIVE | DMTEST)'
+          self.log.warning("Repo key '{key}' {error} - Skipping".format(key = key, error = msg))
+      except Exception as error:
+        self.log.error("Failed during {name} installation: {reason}".format(name = repoName,
+                                                                            reason = error))
+        # If dmtest-python install failed, do not continue any further
+        if (key.upper().find('DMTEST') != -1):
+          sys.exit(1)
+        else:
+          self.log.warning("Manual install of {name} required".format(name = repoName))
+      finally:
+        os.chdir(RUN_DIR)
+
+      self.log.info("Completed installation of {0}\n".format(repoName))
+
+  def installBufioTest(self, repo):
+    """
+    Perform installation steps for the test-bufio repo
+    """
+    # Skip if module is already installed
+    moduleName = 'dm_bufio_test'
+    result = runCommandIgnoringErrors(['lsmod | grep', moduleName])
+    if (result.returncode == 0):
+      self.log.debug("Module {0} is already installed - skipping".format(moduleName))
+      return
+
+    # Obtain the list of remote OS-specific branches
+    # Bufio-test historically only has rhel-8 and rhel-9 branches
+    remoteBranches = []
+    for ref in repo.remotes.origin.refs:
+      if ("rhel" in ref.name):
+        remoteBranches.append(ref.name.replace('origin/', "", 1))
+    self.log.debug("Bufio remote OS-specific branches are: {0}".format(remoteBranches))
+
+    # Determine the rhel release branch to use for installation
+    if (os.path.exists('/etc/fedora-release')):
+      osVersion = 9
+    elif (os.path.exists('/etc/redhat-release')):
+      result = runCommand('cut -d " " -f6 /etc/redhat-release | cut -d"." -f1')
+      osVersion = result.stdout.rstrip()
+    elif (os.path.exists('/etc/centos-release')):
+      result = runCommand('cut -d " " -f4 /etc/centos-release')
+      osVersion = result.stdout.rstrip()
```

Comment:
What if I'm running this on Ubuntu or some other distro?  Perhaps there should be a block that catches that.
```
else:
  raise OSError "Unsupported OS in use"
```
... or something

Reply:
Fixed in latest push.